planner/core/planbuilder_test.go

```diff
@@ -172,3 +185,317 @@ func (s *testPlanBuilderSuite) TestDisableFold(c *C) {
+
+	// index lookup
+	indexLookup := &PhysicalIndexLookUpReader{
+		IndexPlans:     []PhysicalPlan{indexReader},
+		indexPlan:      indexScan,
+		TablePlans:     []PhysicalPlan{tableReader},
+		tablePlan:      tableScan,
+		ExtraHandleCol: col,
+		PushedLimit:    &PushedDownLimit{1, 2},
+	}
+	indexLookup = indexLookup.Init(ctx, 0)
+	c.Assert(checkPhysicalPlanClone(indexLookup), IsNil)
+
+	// selection
+	sel := &PhysicalSelection{Conditions: []expression.Expression{col, cst}}
+	sel = sel.Init(ctx, stats, 0)
+	c.Assert(checkPhysicalPlanClone(sel), IsNil)
+
+	// projection
+	proj := &PhysicalProjection{Exprs: []expression.Expression{col, cst}}
+	proj = proj.Init(ctx, stats, 0)
+	c.Assert(checkPhysicalPlanClone(proj), IsNil)
+
+	// limit
+	lim := &PhysicalLimit{Count: 1, Offset: 2}
+	lim = lim.Init(ctx, stats, 0)
+	c.Assert(checkPhysicalPlanClone(lim), IsNil)
+
+	// sort
+	byItems := []*util.ByItems{{Expr: col}, {Expr: cst}}
+	sort := &PhysicalSort{ByItems: byItems}
+	sort = sort.Init(ctx, stats, 0)
+	c.Assert(checkPhysicalPlanClone(sort), IsNil)
+
+	// topN
+	topN := &PhysicalTopN{ByItems: byItems, Offset: 2333, Count: 2333}
+	topN = topN.Init(ctx, stats, 0)
+	c.Assert(checkPhysicalPlanClone(topN), IsNil)
+
+	// stream agg
+	streamAgg := &PhysicalStreamAgg{basePhysicalAgg{
+		AggFuncs:     aggDescs,
+		GroupByItems: []expression.Expression{col, cst},
+	}}
+	streamAgg = streamAgg.initForStream(ctx, stats, 0)
+	streamAgg.SetSchema(schema)
+	c.Assert(checkPhysicalPlanClone(streamAgg), IsNil)
+
+	// hash agg
+	hashAgg := &PhysicalHashAgg{basePhysicalAgg{
+		AggFuncs:     aggDescs,
+		GroupByItems: []expression.Expression{col, cst},
+	}}
+	hashAgg = hashAgg.initForHash(ctx, stats, 0)
+	hashAgg.SetSchema(schema)
+	c.Assert(checkPhysicalPlanClone(hashAgg), IsNil)
+
+	// hash join
+	hashJoin := &PhysicalHashJoin{
+		Concurrency:     4,
+		UseOuterToBuild: true,
+	}
+	hashJoin = hashJoin.Init(ctx, stats, 0)
+	hashJoin.SetSchema(schema)
+	c.Assert(checkPhysicalPlanClone(hashJoin), IsNil)
+
+	// merge join
+	mergeJoin := &PhysicalMergeJoin{
+		CompareFuncs: []expression.CompareFunc{expression.CompareInt},
+		Desc:         true,
+	}
+	mergeJoin = mergeJoin.Init(ctx, stats, 0)
+	mergeJoin.SetSchema(schema)
+	c.Assert(checkPhysicalPlanClone(mergeJoin), IsNil)
+}
+
+//go:linkname valueInterface reflect.valueInterface
+func valueInterface(v reflect.Value, safe bool) interface{}
+
+func typeName(t reflect.Type) string {
+	path := t.String()
+	tmp := strings.Split(path, ".")
+	return tmp[len(tmp)-1]
+}
+
+func checkPhysicalPlanClone(p PhysicalPlan) error {
+	cloned, err := p.Clone()
+	if err != nil {
+		return err
+	}
+	whiteList := []string{"*property.StatsInfo", "*sessionctx.Context", "*mock.Context", "*types.FieldType"}
+	return checkDeepCloned(reflect.ValueOf(p), reflect.ValueOf(cloned), typeName(reflect.TypeOf(p)), whiteList, nil)
+}
+
+// checkDeepCloned is used to check if v2 is deep cloned from v1.
+// It's modified from reflect.deepValueEqual.
```

Comment:
Can we use `reflect.DeepEqual`?

Reply:
No, since there are some differences between these two functions, for example, if two pointers point the same address, they will pass the `DeepEqual` check while failing in the `DeepClone` check.

Reply:
We need to add this to the comment?

Reply:
Updated. PTAL @XuHuaiyu